examples/js/loaders/GLTFLoader.js

```diff
@@ -704,6 +710,81 @@ THREE.GLTFLoader = ( function () {
 
 	};
 
+
+	/**
+	 * WebP Texture Extension
+	 *
+	 * Specification: https://github.com/KhronosGroup/glTF/tree/master/extensions/2.0/Vendor/EXT_texture_webp
+	 */
+	function GLTFTextureWebPExtension( parser ) {
+
+		this.parser = parser;
+		this.name = EXTENSIONS.EXT_TEXTURE_WEBP;
+		this.isSupported = null;
+
+	}
+
+	GLTFTextureWebPExtension.prototype.loadTexture = function ( textureIndex ) {
+
+		var name = this.name;
+		var parser = this.parser;
+		var json = parser.json;
+
+		var textureDef = json.textures[ textureIndex ];
+
+		if ( ! textureDef.extensions || ! textureDef.extensions[ name ] ) {
+
+			return null;
+
+		}
+
+		var extension = textureDef.extensions[ name ];
+		var source = json.images[ extension.source ];
+		var loader = source.uri ? parser.options.manager.getHandler( source.uri ) : parser.textureLoader;
+
+		return this.detectSupport().then( function ( isSupported ) {
+
+			if ( isSupported ) return parser.loadTextureImage( textureIndex, source, loader );
+
+			if ( ! isSupported && json.extensionsRequired && json.extensionsRequired.indexOf( name ) >= 0 ) {
+
+				throw new Error( 'THREE.GLTFLoader: WebP required by asset but unsupported.' );
+
+			}
+
+			// Fall back to PNG or JPEG.
+			return parser.loadTexture( textureIndex );
```

Comment:
nit: `return null` is better, as it preserves the parsing flow *and* in theory allows one to add a webp *and* a basisu texture replacement, even though the use case is admittedly narrow :)

Reply:
I'd prefer that too, but it's a larger change that I'm nervous to make without more testing. `parser._invokeOne` currently uses the first truthy result — where `Promise<null>` is still truthy — so a method needs to return a real `null` synchronously in order to fall back to the next extension.

https://github.com/mrdoob/three.js/blob/3d8991879c6220684f19a15b16c20b6886d386b1/examples/js/loaders/GLTFLoader.js#L1860-L1873

A straightforward fix would introduce more async indirection to the code, including to things like accessors, so I'd like to wait on that until we're sure it doesn't negatively affect loading time.

Reply:
Yeah, ideally moving to the next extension handler as fallback should be better but as Don explained currently `invokeOne()` doesn't accept `Promise<null>` as the sign of moving to the next handler. The reason why I designed so is if it accepts `Promise<null>` `invokeOne()` can form a long Promise chain and I think it can make the response time very longer.

We need to update or test more `invokeOne()` to resolve this issue. I'd like to suggest this PR goes as is so far, and discussing `invokeOne()` in another issue/PR because I don't want to make this PR bigger and complex.

Reply:
Ah, that makes sense - I didn't realize this return is from a promise wrapper. 👍

Reply:
It may be a bit inefficient and it may not be good to call private method (whose prefix is `_`) but saving boolean indicating WebP is supported in `detectSupport()` and calling `parser._invokeOne()` can invoke other plugins.
```
GLTFTextureWebPExtension.prototype.loadTexture = function ( textureIndex ) {

	if ( ! textureDef.extensions || ! textureDef.extensions[ name ] ||
		this.supported === false ) {

		return null;

	}

	...

	return this.detectSupport().then( function ( isSupported ) {

		...

		// Fall back
		return parser._invokeOne(ext => {return ext.loadTexture && ext.loadTexture(textureIndex)});

	} );

};

GLTFTextureWebPExtension.prototype.detectSupport = function () {

	if ( ! this.isSupported ) {

		this.isSupported = new Promise( function ( resolve, reject ) {

			...

			image.onload = image.onerror = function () {

				const succeeded = image.height === 1;
				this.supported = succeeded;
				resolve( succeeded );

			};

		} );

	}

	return this.isSupported;

};
```